frameworks/moveos-stdlib/src/natives/helpers.rs

```diff
@@ -58,7 +62,22 @@ where
                     },
                 }
             }
-            result
+            if let Err(err) = &result {
+                let status = err.major_status();
+                let vm_status = VMStatus::error(status, Some("".to_string()));
+                match vm_status.keep_or_discard() {
+                    Ok(_) => result,
+                    Err(_) => {
+                        log::error!("Native function execution failed {}", err.to_string());
+                        Ok(NativeResult::err(
```

Comment:
We should convert the discard VMError to a keept VMError, not NativeResult::err

Reply:
The return type of this function is `PartialVMResult`, but the struct `PartialVMError` does not have the `Location` information in it:

```rust
struct PartialVMError_ {
    major_status: StatusCode,
    sub_status: Option<u64>,
    message: Option<String>,
    exec_state: Option<ExecutionState>,
    indices: Vec<(IndexKind, TableIndex)>,
    offsets: Vec<(FunctionDefinitionIndex, CodeOffset)>,
}
```